slam-libraries/viam-orb-slam3/orbslam_server_v1_main.cc

```diff
@@ -170,6 +169,7 @@ int main(int argc, char **argv) {
         SLAM->GetAtlas()->ChangeMap(largestMap);
     }
 
+    slamService.SetSlam(SLAM.get());
```

Comment:
i saw this, ty for the cleanup!

Reply:
We should also call `slamService.SetSlam(nullptr)` prior to the end of this function. Otherwise, if `SLAM` gets destroyed before `server`, a call to `GetInternalState` may try to access invalid memory.

Reply:
I've added it. Please let me know if that is sufficient to make it safe.